chrome/content/coniks/coniks.js

```diff
@@ -0,0 +1,90 @@
+this.EXPORTED_SYMBOLS = ["coniks"];
+
+var { interfaces: Ci, utils: Cu, classes: Cc } = Components;
+
+Cu.import("resource:///modules/imServices.jsm");
+Cu.import("resource:///modules/imXPCOMUtils.jsm");
+Cu.import("resource:///modules/jsProtoHelper.jsm");
+Cu.import("chrome://otr/content/coniks/context.js");
+Cu.import("chrome://otr/content/helpers.js");
+Cu.import("chrome://otr/content/otr.js");
+
+function accountname(acc) {
+  let account = acc.normalizedName;
+  let protocol = acc.protocol.normalizedName;
+  return account+"@"+protocol;
+}
+
+var coniks = {
+  _isEnable: true,
+  get Enabled() { return coniks._isEnable; },
+
+  init: function() {
+    coniks.setConfig();
+    this._contexts = new Map();
+    for (let acc of helpers.getAccounts()) {
+      coniks._contexts.set(accountname(acc), new AccountContext({
+        account: acc.normalizedName,
+        protocol: acc.protocol.normalizedName
+      }));
+    }
+
+    coniks.loadFiles().then(function() {
+      Services.obs.addObserver(coniks, "account-added", false);
+    }).catch(function(err) { throw err; });
+  },
+
+  observe: function(aObject, aTopic, aMsg) {
+    switch(aTopic) {
+    case "account-added":
+      coniks.onAccountCreated(aObject);
+      break;
+    }
+  },
+
+  destroy: function() {
+    Services.obs.removeObserver(coniks, "account-added");
+  },
+
+  onAccountCreated: function(acc) {
+    let account = acc.normalizedName;
+    let protocol = acc.protocol.normalizedName;
+    let fingerprint = otr.privateKeyFingerprint(account, protocol);
+    coniks._contexts.set(accountname(acc), new AccountContext({
+      account: account,
+      protocol: protocol,
+      fingerprint: fingerprint
+    }));
+    ConiksContexts.storeToFile(coniks._contexts);
+  },
+
+  setConfig: function() {
+    let branch = "extensions.otr.coniks.";
+    let serverConfig = {
+      serverAddress: "111.221.102.190",
+      serverPort: 3000
+    };
+    let defaults = Services.prefs.getDefaultBranch(branch);
+    Object.keys(serverConfig).forEach(function(key) {
+      defaults.setCharPref(key, serverConfig[key]);
+    });
+  },
+
+  loadFiles: function() {
+    return Promise.all([
```

Comment:
Promise.all is unnecessary since you're only resolving one promise (which, isn't a promise yet anyways)